torch/csrc/distributed/spmd/event_handler_impl.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
```

Comment:
the original design was intended to decouple the engine from DDP and make it generic, in case there are other non-DDP use cases in the future and in case researcher would like to try fancy ideas.